### DISCUSSION TITLE
How can I access query parameter from the callback?

### DISCUSSION THREAD

**Discussion by janhgit** [nextauthjs/next-auth#4629] [Help]
Hey everyone! I am using last.fm as a custom provider and I am kind of lost 🤦‍♂️I am probably just overseeing something.
I tried following [this](https://github.com/nextauthjs/next-auth/discussions/3156) but i did get only more confused

So here's my issue:
Everything works fine until after the user authorizes the application.
last.fm sends the user back to the callback `api/auth/callback/lastfm` , with a token. so its `api/auth/callback/lastfm?token=kadklsakdlaBlaBlaBla`

My problem is that I instead get sent to `api/auth/signin?error=OAuthCallback`

![grafik](https://user-images.githubusercontent.com/81878834/170787590-60a1ca7c-0f73-4de3-97f5-5396f8bc9dc3.png)
You can see it here

I need to access this token though, to continue with the OAuth cycle and make a new call for a session ( don't need help with that; I only need help with accessing the token in the callback)

Thanks for the help IA and have a great evening 😺

**Comment by niall-byrne** [comment]
I just ran into this problem, and wanted to share my workaround (although I'm sure you've since figured something out, maybe this will help someone else?)

This is a last.fm specific issue, as it is not quite 100% oauth compliant.

Creating a simple redirect endpoint that maps the non-standard `token` query parameter variable to a oauth standard one like `id_token` worked best for me.

As a sequence:
- call the lastfm authorization endpoint telling it to use the new custom callback api endpoint
- the lastfm endpoint then calls the custom callback endpoint with `token`, and it is 302 redirected (not 301) to the actual next-auth endpoint where `token` has been mapped to a oauth standard query string variable name (such as `id_token`)
- now the query parameter is accessible and authentication can proceed
